### PR TITLE
new API end point for project categories

### PR DESCRIPTION
**Pull request by ndrnour**
---
### Your checklist for this pull request
Please review the [contributing guidelines](https://github.com/Catrobat/Catroweb-Symfony/blob/develop/.github/contributing.md) of this repository.

- [ ] Include the name and id of the Jira ticket in the PR’s title eg.: `SHARE-666 The devils ticket`
- [ ] Choose the proper base branch (*develop*)
- [ ] Confirm that the changes follow the project’s coding guidelines
- [ ] Verify that the changes generate no warnings and errors 
- [ ] Verify to commit no other files than the intentionally changed ones
- [ ] Include reasonable and readable tests verifying the added or changed behavior
- [ ] Verify that all tests are passing, if not please state the test cases in the [section](#Tests) below
- [ ] Perform a self-review of the changes
- [ ] Stick to the project’s git workflow (rebase and squash your commits)
- [ ] Verify that your changes do not have any conflicts with the base branch
- [ ] Put your ticket into the `Code Review` section in [Jira](https://jira.catrob.at/)
- [ ] Post a message in the *#catroweb* [Slack channel](https://catrobat.slack.com) and ask for a code reviewer

### Additional Description
`TODO: Add additional information that is not in your commit-message here`

### Tests - additional information
`TODO: add additional information about testruns here`

### CODE REVIEW

**Review by dmetzner — CHANGES_REQUESTED** [review]
Points to discuss:

1) Returning more than 100k projects at once seems to be an overkill for most use cases.
I'd recommend adding optional additional limit & offset parameters for pagination.
This also implicitly allows API end-users to query for all available categories (limit=0 call)
 
   - int `limit`. E.g. `limt=20` (default)
   - int `offset` E.g. `offset=0` (default)
 
 2) We have a lot of categories. And some of them are not very optimized for performance (E.g. recommendations). I would add an additional optional parameter called categories, to filter only the categories that are needed.
 
    - array `categories`. E.g.: `['recent', 'newest']` (default=all categories)
   
   @HCrane @ndrnour 
   
   @m-herold Is such an API endpoint for IOS maybe also interesting?

File: catroweb.yaml [hunk 1]
```diff
@@ -1427,6 +1455,26 @@ components:
...
 
+    ProjectsCategoryResponse:
+      type: array
+      items:
+        $ref: '#/components/schemas/ProjectsCategory'
+
+    ProjectsCategory:
+      type: object
+      properties:
+        id:
```
Comment (dmetzner) [inline catroweb.yaml]: the `id` can be removed here, the `name` is already a unique identifier.

Comment (HCrane) [reply catroweb.yaml]: The id will be the name of the category in English. the name we return shall be the translation to the given language.

**Review by HCrane — COMMENTED** [review]
As for limit and offset Nur and I discussed that we will enshrine our default limits (20/0) and not dynamic. for pagination or more specific projects we already supply the needed apis.

File: catroweb.yaml (hunk 1 above)
Comment (dmetzner) [reply catroweb.yaml]: Then I would rather choose names that give a clear indication of this.
id is name, and name is translated category title sounds a bit unintuitive to me.
At least it should be explained in the specification

**Comment by dmetzner** [comment]
Then this should also be clearly documented in the specification

**Comment by wslany** [comment]
Any news here?

**Comment by HCrane** [comment]
Depending on how much time uni takes tomorrow ill implement the changes and also update the API on the main repo tomorrow.

**Comment by HCrane** [comment]
@ndrnour could you quickly check the changes I made if they are appropriate?

**Review by dmetzner — APPROVED** [review]

**Pull request merged**